ambry-tools/src/main/java/com.github.ambry/clustermap/HelixVcrPopulateTool.java

```diff
@@ -0,0 +1,183 @@
+        parser.accepts("dest").withRequiredArg().describedAs("dest zk and cluster name").ofType(String.class);
+
+    OptionSet options = parser.parse(args);
+
+    String destZkString = options.valueOf(destOpt).split(SEPARATOR)[0];
+    String destClusterName = options.valueOf(destOpt).split(SEPARATOR)[1];
+    if (!destClusterName.contains("VCR")) {
+      System.out.println("dest should be a VCR cluster.(VCR string should be included)");
+      return;
+    }
+
+    if (options.has(createClusterOpt)) {
+      System.out.println("Creating cluster: " + destClusterName);
+      createCluster(destZkString, destClusterName);
+    }
+
+    if (options.has(updateClusterOpt)) {
+      System.out.println("Updating cluster: " + destClusterName);
+      String srcZkString = options.valueOf(srcOpt).split(SEPARATOR)[0];
+      String srcClusterName = options.valueOf(srcOpt).split(SEPARATOR)[1];
+      boolean dryRun = options.has(dryRunOpt);
+      updateResourceAndPartition(srcZkString, srcClusterName, destZkString, destClusterName, dryRun);
+    }
+    System.out.println("Done.");
+  }
+
+  /**
+   * Create a helix cluster with given information.
+   * @param destZkString the cluster's zk string
+   * @param destClusterName the cluster's name
+   * @return false if cluster already exist, otherwise true.
+   */
+  static boolean createCluster(String destZkString, String destClusterName) {
+    HelixZkClient destZkClient =
+        DedicatedZkClientFactory.getInstance().buildZkClient(new HelixZkClient.ZkConnectionConfig(destZkString));
+    destZkClient.setZkSerializer(new ZNRecordSerializer());
+    HelixAdmin destAdmin = new ZKHelixAdmin(destZkClient);
+    if (destAdmin.getClusters().contains(destClusterName)) {
+      System.out.println("Create cluster failed. " + destClusterName + " already exist.");
+      return false;
+    }
+    ClusterSetup clusterSetup = new ClusterSetup(destZkString);
+    clusterSetup.addCluster(destClusterName, true);
+
+    // set ALLOW_PARTICIPANT_AUTO_JOIN
+    HelixConfigScope configScope = new HelixConfigScopeBuilder(HelixConfigScope.ConfigScopeProperty.CLUSTER).
+        forCluster(destClusterName).build();
+    Map<String, String> helixClusterProperties = new HashMap<>();
+    helixClusterProperties.put(ZKHelixManager.ALLOW_PARTICIPANT_AUTO_JOIN, String.valueOf(true));
+    destAdmin.setConfig(configScope, helixClusterProperties);
+    // set PersistBestPossibleAssignment
+    ConfigAccessor configAccessor = new ConfigAccessor(destZkClient);
+    ClusterConfig clusterConfig = configAccessor.getClusterConfig(destClusterName);
+    clusterConfig.setPersistBestPossibleAssignment(true);
+    configAccessor.setClusterConfig(destClusterName, clusterConfig);
+    System.out.println("Cluster " + destClusterName + " create done.");
+    return true;
+  }
+
+  /**
+   * Update dest cluster information based on src cluster. Dest cluster resource will be recreated if src cluster has any change.
```

Comment:
minor question: is `recreated` automatically? In other words, any changes in src cluster will trigger changes in dest cluster automatically?

Reply:
Never mind, I understand what you mean here by reading following code. However, I feel like it could be better to change the description as:
`Dest cluster resource will be recreated if it mismatches that in src cluster.`